website/docs/bestpractices/analytics/spark-best-practices.md

```diff
@@ -0,0 +1,156 @@
+
+### Custom Schedulers
+
+Custom schedulers enhance Kubernetes’ native scheduling capabilities by providing advanced features tailored for batch and high-performance computing workloads. Custom schedulers enhance resource allocation by optimizing bin-packing and offering scheduling tailored to specific application needs. Here are popular custom schedulers for running Spark workloads on Kubernetes.
+    * [Apache Yunikorn](https://yunikorn.apache.org/)
+    * [Volcano](https://volcano.sh/en/)
+
+Advantages of leveraging custom schedulers like Yunikorn.
+        * It provides hierarchical queue system and configurable policies allow for complex resource management.
+        * Gang Scheduling
+
+### How will Yunikorn and Karpenter work together?
+
+Karpenter focuses on node provisioning and scaling, while custom schedulers like Yunikorn, brings in application awareness for scheduling.
+
+Yunikorn manages the advanced application scheduling aspects like queue management, resource fairness, gang scheduling.Once Yunikorn schedules the pods, based on application awareness, Karpenter will then fulfil needs of pending pods, spawned by Yunikorn.
+
+
+## Storage Best Practices
+### Node Storage
+By default, the EBS root volumes of worker nodes is set to 20GB. Spark Executors use local storage for temporary data like shuffle data, intermediate results and temporary files. Default storage of 20GB of root volume attached to worker nodes can be limiting in size and performance. Consider the following options to cater to your performance and storage size requirements.
+        * Expand the root volume capacity.
+        * Configure high-performance storage with better I/O and latency.
+        * Mount additional volumes on worker nodes for temporary data storage.
+        * Leverage dynamically provisioned PVCs that can be attached directly to executor pods.
+
+### Reuse PVC
+This option allows reusing PVCs associated with Spark executors even after the executors are terminated (either due to consolidation activity or preemption in case of Spot instances).
+
+This allows for preserving the intermediate shuffle data and cached data on the PVC. When Spark requests new executor pod to replace the terminated one, the system attempts to reuse an existing PVC that belonged to terminated executor. This option can be enabled by the following configuration:
+
+`spark.kubernetes.executor.reusePersistentVolume=true`
+
+### External Shuffle services like `Apache Celeborn`
+Leverage external shuffle service to decouple compute and storage, allowing spark executors to write data to external shuffle service instead of local disks, reducing the risk of data loss and data re-computation due to executor termination/consolidation.
```

Comment:
```suggestion
External shuffle services like Apache Celeborn decouple compute and storage, allowing Spark executors to write data to an external service instead of local disks. This reduces the risk of data loss and re-computation due to executor termination or consolidation.

```